### PR TITLE
Fix handling of format strings in custom script types

### PR DESCRIPTION
**Pull request by wilbowma**
This helps address #37 

1. Some tools output the "TeX" format as "tex", so normalize the type string.
2. Differentiate between "TeX", aka "inline-TeX", and "TeX; mode=display", aka "TeX".

I've tested it on the following webpage and it seems to prerender properly: https://williamjbowman.com/blog/2017/03/24/what-even-is-compiler-correctness/

### CODE REVIEW

**Comment by pkra** [comment]
Thanks. 

Can you add some tests (and fix the broken ones)?

**Comment by pkra** [comment]
Though I think this would be better handled in the pre-processors themselves (not `main`).

**Comment by wilbowma** [comment]
Sure! I'm pretty new to JavaScript hacking so if I've done things in a less than ideal way, let me know.

**Comment by wilbowma** [comment]
I'm a bit confused. The tests seems to give equations in `\\[ \\]` the format `math/tex`. However, in my webpages MathJax renders anything in `math/tex` (without `; mode=display`) as an inline equation in my webpages.

File: lib/main.js [hunk 1]
```diff
@@ -113,9 +113,30 @@ exports.mjpage = function(htmlstring, configOptions, typesetOptions, callback) {
         window.ascii.config.doc = document;
         window.ascii.PreProcess();
     }
+    //normalize type strings
+    const scripts = document.querySelectorAll('script[type^="math/"]');
+    for (let i = 0; i < scripts.length; i++) {
+        const script = scripts[i];
+        let format = script.getAttribute("type").toLowerCase().slice(5);
+        if (format === "inline-tex") {
```
Comment (pkra) [inline lib/main.js]: What's the rationale for this check? It seems redundant.

File: lib/main.js [hunk 2]
```diff
@@ -113,9 +113,30 @@ exports.mjpage = function(htmlstring, configOptions, typesetOptions, callback) {
...
+            format = "math/inline-TeX";
+            // via https://github.com/mathjax/MathJax/blob/master/unpacked/jax/input/TeX/jax.js#L2168
+        } else if (format.match(/tex((;|\s*|\n*)mode\s*=\s*display)?/i)) {
```
Comment (pkra) [inline lib/main.js]: I'm no regex expert, but I believe this will be true for `math/tex` as well. I would suggest to simply compare against `math/tex; mode=display`.

File: lib/main.js [hunk 3]
```diff
@@ -113,9 +113,30 @@ exports.mjpage = function(htmlstring, configOptions, typesetOptions, callback) {
...
+            format = "math/TeX";
+        } else if (format === "mathml") {
+            format = "math/MathML";
+        } else if (format === "mathml-block") {
+            format = "math/MathML-block";
+        } else if (format === "asciimath") {
+            format = "math/AsciiMath";
+        } else {
+            console.log("Unknown format " + format);
```
Comment (pkra) [inline lib/main.js]: This is a good point but you should really remove that node (it will still get processed later). But removing nodes from nodelists requires conversion to an array. Makes me wonder...

**Comment by pkra** [comment]
Thanks again for the PR -- it's very helpful. I see a couple of problems and I'm starting to favor a simpler solution.

1. only support the original MathJax script types for TeX input
  * Rationale: the type for asciimath is ok and it seem unlikely that anybody using MathML would pre-process it into script tags (and then we'd need to identify the display type which is not ideal).
2. fix a problem that your test shows: we capture too many script tags, i.e., https://github.com/pkra/mathjax-node-page/blob/master/lib/main.js#L118 should match precisely our script types.

If you want to give that a go, let me know. If not, I'm happy to do it myself.

**Comment by pkra** [comment]
FYI I pushed a continuation of your branch to https://github.com/pkra/mathjax-node-page/tree/wilbowma.

**Comment by pkra** [comment]
These commits were merged via #43. Thanks again for making the initial PR.
